kythe/go/extractors/golang/golang.go

```diff
@@ -435,11 +435,20 @@ func (p *Package) addFiles(cu *apb.CompilationUnit, root, base string, names []s
 			path = filepath.Join(base, name)
 		}
 		trimmed := strings.TrimPrefix(path, root+"/")
+		vn := &spb.VName{
+			Corpus: p.ext.Corpus,
+			Path:   trimmed,
+		}
+		if vn.Corpus == "" {
+			vn.Corpus = p.VName.Corpus
+			components := strings.SplitN(vn.Path, string(filepath.Separator), 2)
+			vn.Path = strings.TrimPrefix(components[1], vn.Corpus+"/")
+			if components[0] != "src" {
```

Comment:
Is this a known pattern for go?

Reply:
Yes, it's the structure of a GOPATH.

Reply:
Unfortunately it's not quite that simple, because different repository hosts use different structures. kythe/kythe/go/extractors/govname has the logic that the Go tool uses—or at least that it used when I wrote that (it may need updated).

Reply:
(specifically: Knowing which part is the actual package name from the file path alone is tricky)

Reply:
For custom repository hosts, the corpus is the top-level domain (e.g. "k8s.io).  I plan to expand the config to better support these cases.

Reply:
Could you add a comment to that effect? Now that you describe it it makes sense, but it wasn't obvious that it does the right thing from the code alone.